### ISSUE TITLE
While the loss is going down, the prediction accuracy is going down too

### ISSUE THREAD

**Issue by mtichavsky** [simon-smida/KNN-project#18]
Analyzing my batch job files

```shell
cat <batch_job_epoch_out> | grep "prediction accuracy" | cut -d " " -f 8 | awk '{s+=$1} END {print s}'
```

leads to values for epochs

1: 0.215
2: 0.17125
3: 0.13375
4: 0.145

I might have an issue somewhere in the code (at least in the prints during training).

**Comment by mtichavsky** [comment]
Preliminary evaluation (1k samples):

1 iteration
Model achieves EER = 9.00% at threshold 0.60 on the first 1000 samples of the test split.
Model achieves minDCF = 0.02 at threshold 0.68 on the first 1000 samples of the test split.

2 iterations
Model achieves EER = 4.40% at threshold 0.60 on the first 1000 samples of the test split.
Model achieves minDCF = 0.01 at threshold 0.67 on the first 1000 samples of the test split.

3 iterations
Model achieves EER = 5.00% at threshold 0.48 on the first 1000 samples of the test split.
Model achieves minDCF = 0.01 at threshold 0.55 on the first 1000 samples of the test split.

4 iterations
Model achieves EER = 3.00% at threshold 0.49 on the first 1000 samples of the test split.
Model achieves minDCF = 0.009 at threshold 0.551 on the first 1000 samples of the test split.

**Comment by mtichavsky** [comment]
Fixed by #28
